mindmeld/components/role_classifier.py

```diff
@@ -105,24 +105,24 @@ def fit(self,
         new_hash = self._get_model_hash(model_config, queries)
         cached_model_path = self._resource_loader.hash_to_model_path.get(new_hash)
 
-        if incremental_timestamp and cached_model_path:
-            logger.info("No need to fit. Previous model is cached.")
-            if load_cached:
-                # load() sets self.ready = True
-                self.load(cached_model_path)
-                return True
-            return False
-
         # These examples and labels are flat lists, not
         # a ProcessedQueryList.Iterator
         examples, labels = self._get_examples_and_labels(queries)
 
         if examples:
             # Build roles set
-            self.roles = set()
             for label in labels:
```

Comment:
This operation is more efficient if you simply use `self.roles.update(labels)` rather than the for loop.

Reply:
Fixed!